test/smtcallback.js

```diff
@@ -34,8 +34,19 @@ function expectErrors (errors, expectations) {
     if (errors[i].includes('Error trying to invoke SMT solver') || expectations[i].includes('Error trying to invoke SMT solver')) {
       continue;
     }
-    expectations[i] = expectations[i].replace('happens here.', 'happens here');
-    if (!errors[i].includes(expectations[i])) {
+    // Expectations containing counterexamples might have many '\n' in a single line.
+    // When these are read, they come as '\\n', whereas the actual error from the compile has '\n'.
+    // Therefore we need to replace '\\n' by '\n' in the expectations.
+    // Function `replace` only replaces the first occurrence, and `replaceAll` is not standard yet.
+    // Replace all '\\n' by '\n' via split & join.
+    expectations[i] = expectations[i].split('\\n').join('\n');
```

Comment:
This is needed here for the case where we actually want to compare the counterexamples